src/core/math/vec2.js

```diff
@@ -407,6 +409,57 @@ class Vec2 {
         return this;
     }
 
+    /**
+     * Rotate a vector by Euler angle.
+     *
+     * @param {number} degrees - The number to rotate by Euler angle.
+     * @returns {Vec2} Self for chaining.
+     * @example
+     * const v = new pc.Vec2(0, 10);
+     *
+     * v.rotate(45); // rotates by 45 degrees
+     *
+     * // Outputs [7.071068.., 7.071068..]
+     * console.log("Vector after rotation is: " + v.toString());
+     */
+    rotate(degrees) {
+        const angle = Math.atan2(this.x, this.y) + (degrees * math.DEG_TO_RAD);
+        const len = Math.sqrt(this.x * this.x + this.y * this.y);
+        this.x = Math.sin(angle) * len;
+        this.y = Math.cos(angle) * len;
+        return this;
+    }
+
+    /**
+     * Returns the Euler angle of the specified 2-dimensional vector.
+     *
+     * @returns {number} The Euler angle of the specified 2-dimensional vector.
+     * @example
+     * const v = new pc.Vec2(6, 0);
+     * const angle = v.angle();
+     * // Outputs 90..
+     * console.log("The angle of the vector is: " + angle);
+     */
+    angle() {
```

Comment:
Question: should this be a function or a getter?

Reply:
I believe it should be a function, to stay consistent with similar methods, such as: length()

Reply:
That's a good and fair point. But I'm wondering whether length, for example, should also be a getter these days. If we *do* decide to leverage getters in the math API, that's for another PR... 😄